rest/client/methods.go

```diff
@@ -332,11 +332,14 @@ func (c *communicatorImpl) GetTaskPatch(ctx context.Context, taskData TaskData)
 	}
 	info.setTaskPathSuffix("git/patch")
 	resp, err := c.retryRequest(ctx, info, nil)
-
 	if err != nil {
 		return nil, errors.Wrapf(err, "could not get patch for %s", taskData.ID)
 	}
+	defer resp.Body.Close()
 
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.Errorf("could not find patch for task with id '%s'", taskData.ID)
```

Comment:
I'd be more explicit here that the reason we're erroring is because we didn't get a 200 OK.

Reply:
yep, added the status code.

Reply:
lgtm